src/Stratis.Bitcoin/Base/TipsManager.cs

```diff
@@ -132,6 +144,16 @@ public ChainedHeader GetLastCommonTip()
             return this.lastCommonTip;
         }
 
+        /// <inheritdoc />
+        public void CommitTipPersisted(ITipProvider provider, HashHeightPair tip)
+        {
+            ChainedHeader header = this.concurrentChain.GetBlock(tip.Hash);
+            if (tip == null)
+            {
+                throw new ArgumentException(string.Format("Invalid Hash: {0} not found in current chain", tip.Hash));
+            }
```

Comment:
Aren't you forgetting a call here?

Reply:
yes the other CommitTipPersisted overload